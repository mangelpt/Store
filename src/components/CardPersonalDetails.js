import styled from "@emotion/styled";
import  {Input} from "./UI/Input"
import photo from "./person.png"

const CardPersonal = styled.div`
  background-color: #ffff;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.03);
  border-radius: 20px;
  padding: 16px 18px 18px 16px;
  width: 315px;
  display: flex;
  flex-direction: row;
  gap: 15px;
  margin:0;
  padding: unset;
  margin-top:10px;
  align-self: center;
  form {
    margin-top:18px;
    width: 165px;
    display: flex;
    flex-direction: column;
  }

  input {
    font-weight: 400;
    font-size: 18px;
    line-height: 23px;
    color: #333333;
    border: none;
    background: none;
    outline: none;
  }

  input[name="name"] {
    font-weight: 600;
  }

  input::placeholder {
    opacity: 0.5;
  }

  div {
    margin: 7px 0 10px 0;
    opacity: 0.2;
    border-top: 0.5px solid #000000; 
  }

  img {
    width: 91px;
    height: 100px;
    margin-top:18px;
    margin-left:16px;
  }
`;


export const TextArea = styled.textarea`
  width: 176px;
  min-height: 55px;
  color: #333333;
  background: #ffffff;
  border: none;
  box-sizing: border-box;
  ::placeholder {
    color: black;
    opacity: 0.5;
  }
  font-family: Source Sans Pro;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 23px;
  border: 0;
  outline: none;
`;

export function CardPersonalDetails(props) {
  return (
    <CardPersonal>
      <img src={`${props.avatar_url|| photo}`} alt="profile" />
      <form>
        <Input
          type="text"
          name="name"
          value={props.name}
          placeholder="name"
          disabled="disabled"
        />
        <div />
        <Input
          type="email"
          name="email"
          value={props.email}
          placeholder="email"
          disabled="disabled"
        />
        <div />
        <Input
          type="number"
          name="phone"
          value={props.phone}
          placeholder="phone"
          disabled="disabled"
        />
        <div />
        <TextArea
          type="text"
          name="address"
          value={props.address}
          placeholder="address"
          disabled="disabled"
        />
      </form>
    </CardPersonal>
  );
}