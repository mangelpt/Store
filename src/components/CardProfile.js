import styled from "@emotion/styled"
import  {CardInput} from "./UI/CardInput"
import camera from "./Vector.svg"
import { Lbl } from "./UI/Labels"
import { TextArea } from "./UI/CardInput"
import {ContainerInput} from "./UI/CardInput"

const ContainerForm = styled.form`
    display: flex;
    flex-direction: row;
    gap: 4px;
    width: 314px;
    height: 312px;
    background-color:white;
    background: #FFFFFF;
    box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.03);
    border-radius: 20px;
`

const ContainerInfo = styled.div`
    display: flex;
    flex-direction: column;
    gap:8px;
    width: 214px;
    height: 300px;
    background-color:white;
    padding-top:12px;
    border-radius: 20px;
`

const ConatinerFile = styled.div`
    text-align: center;
    margin: 0 auto;
    height: 80px;
    width: 72px;
    background: #F6F6F9;
    border-radius: 10px;
    margin-top:12px;
    padding-top:12px;
    display: flex;
    flex-direction: column;

    label {
        color: #B8B8BB;
    }

    input[type="file"]{
        display: none;
    }
`
export function CardProfile() {


  return (
    <>
        <ContainerForm>
            <ConatinerFile>
               <label for="file">
                    <img src={camera} alt="imagecamera" />
                    Upload image
               </label>
               <input id="file" type="file" name="file" />
            </ConatinerFile>
            <ContainerInfo>
                <CardInput 
                            textlabel="Name"
                            name="name"
                            type="text"
                            placeholder="Margarita">
                </CardInput> 
                <CardInput 
                            textlabel="Email"
                            name="email"
                            type="email"
                            placeholder="margarita@gmail.com">
                </CardInput> 
                <CardInput 
                            textlabel="Phone"
                            name="phone"
                            type="nuber"
                            placeholder="987654321">
                </CardInput> 
                <ContainerInput>
                    <Lbl>Address</Lbl>
                    <TextArea type="text"
                        name="address"
                        placeholder="Calle el rosal, urb. el Jardin"
                        maxLength="140"
                    >
                    </TextArea>
                </ContainerInput>
            </ContainerInfo>
        </ContainerForm>
    </>
  )
}