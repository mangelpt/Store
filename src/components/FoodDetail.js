import styled from "@emotion/styled";

export function FoodDetail(props) {

  const Container = styled.div`
    width: 75vw;
    display: flex;
    flex-direction: column;
    align-items: center;
  `;

  const ContainImage = styled.div`
    width: 241px;
    height: 241px;
    border-radius: 50%; 
    background: url(${props.picture_url}); 
    background-size: cover; 
  `;

  const NameDish = styled.p`
    width: 100%;
    height: fit-content;
    font-family: "Source Sans Pro";
    font-style: normal;
    font-weight: 600;
    font-size: 28px;
    line-height: 35px;
    text-align: center;
    color: #333333;
    margin-top: 50px;
  `;

  const Price = styled.p`
    width: 80px;
    height: 35px;
    font-family: "Source Sans Pro";
    font-style: normal;
    font-weight: 600;
    font-size: 28px;
    line-height: 35px;
    text-align: center;
    color: #FA4A0C;
    margin-top: 10px;
  `;

  const TitleDescription = styled.p`
    width: 89px;
    height: 23px;
    font-family: "Source Sans Pro";
    font-style: normal;
    font-weight: 600;
    font-size: 18px;
    line-height: 23px;
    color: #333333;
    margin-top: 27px;
    `;

  const Description = styled.p`
    width: 297px;
    height: fit-content;
    font-family: "Source Sans Pro";
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 20px;
    color: #333333;
    margin-top: 4px;
  `;

  return (
    <Container>
      <ContainImage></ContainImage>
      <NameDish>{props.name}</NameDish>
      <Price>{`$${props.price}`}</Price>
      <div>
        <TitleDescription>Description</TitleDescription>
        <Description>{props.description}</Description>
      </div>
    </Container>
  );
}