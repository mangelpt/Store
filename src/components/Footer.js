import styled from '@emotion/styled'
import React from 'react'
import { NavLink } from 'react-router-dom'
import { ButtonWhite } from './UI/Buttons'
import { ArrowIconRight, HistoryIcon, HomeIcon, UserIcon } from './UI/Icons'

const StyledFooter = styled.footer`
width: 75vw;
margin: 10px auto;
height: fit-content;
display: flex;
justify-content:space-between;
svg{
  width: 30px;
  color: gray;
}
`;

export const Footer = () => {
  return (
    <>
      <ButtonWhite> History <ArrowIconRight /> </ButtonWhite>
      <StyledFooter>
        <NavLink to="/home">
          <HomeIcon />
        </NavLink>
        <NavLink to="/user">
          <UserIcon />
        </NavLink>
        <NavLink to="/history">
          <HistoryIcon />
        </NavLink>
      </StyledFooter>
    </>
  )
}
