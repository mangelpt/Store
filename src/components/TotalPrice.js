import styled from "@emotion/styled";

export function TotalPrice({ pricetotal }) {

  const ContainerPrice = styled.div`
    width: 75vw;
    height: fit-content;
    display: flex;
    align-items: center;
    margin: auto;
    justify-content: space-between;
    p{
      font-family: "Source Sans Pro";
      font-style: normal;
      font-weight: normal;
      font-size: 18px;
      line-height: 23px;
      color: #333333;
    }
    label{
      font-family: "Source Sans Pro";
      font-style: normal;
      font-weight: 600;
      font-size: 28px;
      line-height: 35px;
      color: #333333;
    }
  `;

  return (
    <ContainerPrice>
      <p>Total:</p>
      <label>{`$${pricetotal}`}</label>
    </ContainerPrice>
  );
};