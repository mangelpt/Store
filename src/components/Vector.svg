<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M20.75 15.75V3.25H3.25V15.75H20.75ZM0.75 1.99125C0.75 1.30625 1.30625 0.75 1.99125 0.75H22.0088C22.6938 0.75 23.25 1.30625 23.25 1.99125V22.0088C23.2497 22.3378 23.1188 22.6534 22.8861 22.8861C22.6534 23.1188 22.3378 23.2497 22.0088 23.25H1.99125C1.66215 23.2497 1.34663 23.1188 1.11392 22.8861C0.881211 22.6534 0.750331 22.3378 0.75 22.0088V1.99125ZM12 12C12.663 12 13.2989 11.7366 13.7678 11.2678C14.2366 10.7989 14.5 10.163 14.5 9.5C14.5 8.83696 14.2366 8.20107 13.7678 7.73223C13.2989 7.26339 12.663 7 12 7C11.337 7 10.7011 7.26339 10.2322 7.73223C9.76339 8.20107 9.5 8.83696 9.5 9.5C9.5 10.163 9.76339 10.7989 10.2322 11.2678C10.7011 11.7366 11.337 12 12 12ZM12 14.5C10.6739 14.5 9.40215 13.9732 8.46447 13.0355C7.52678 12.0979 7 10.8261 7 9.5C7 8.17392 7.52678 6.90215 8.46447 5.96447C9.40215 5.02678 10.6739 4.5 12 4.5C13.3261 4.5 14.5979 5.02678 15.5355 5.96447C16.4732 6.90215 17 8.17392 17 9.5C17 10.8261 16.4732 12.0979 15.5355 13.0355C14.5979 13.9732 13.3261 14.5 12 14.5ZM4.5 4.5H7V7H4.5V4.5ZM4.5 18.25V20.75H19.5V18.25H4.5Z" fill="#B8B8BB"/>
</svg>
