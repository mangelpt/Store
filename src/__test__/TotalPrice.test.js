import React from "react";
import renderer from "react-test-renderer";
import { TotalPrice } from "../components/TotalPrice";
import {  render } from "@testing-library/react";
describe("Styled Footer", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TotalPrice />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

