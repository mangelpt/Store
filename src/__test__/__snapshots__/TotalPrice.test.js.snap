// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Footer renders correctly 1`] = `
<div
  className="css-ca2rxh"
>
  <p>
    Total:
  </p>
  <label>
    $undefined
  </label>
</div>
`;
